parity/cli/usage.rs

```diff
@@ -312,9 +317,16 @@ macro_rules! usage {
 			pub fn parse<S: AsRef<str>>(command: &[S]) -> Result<Self, ArgsError> {
 				let raw_args = RawArgs::parse(command)?;
 
+				// Invalid configuration pattern `mix_peers` > `max_peers`
+				if let (Some(max_peers), Some(min_peers)) = (raw_args.arg_max_peers, raw_args.arg_min_peers) {
+					if min_peers > max_peers {
+						return Err(ArgsError::PeerConfiguration)
+					}
+				} 
+
 				// Skip loading config file if no_config flag is specified
-				if raw_args.flag_no_config {
-					return Ok(raw_args.into_args(Config::default()));
+				else if raw_args.flag_no_config {
```

Comment:
This `else` is fishy, why do we ignore `no-config` flag if max_peers or min_peers are not provided?

Reply:
Yes, I also think this `else` is a mistake.

Reply:
Yepp, you’re right it should be s/else if/if.

I made a faulty assumption that first if statement catches only errors!